revocation/revocation.go

```diff
@@ -69,7 +70,7 @@ type Validator interface {
 // revocation is an internal struct used for revocation checking
 type revocation struct {
 	ocspHTTPClient   *http.Client
-	crlHTTPClient    *http.Client
+	crlFetcher       crlutil.Fetcher
```

Comment:
Assuming we are taking `crlFetcher` instead of `httpClinet` because we want to support cache. IMO it would good if we can cache also OCSP responses. Please open an issue for this

qq: Will design change if we want to cache ocsp response?

Reply:
Yes, currently we cache the `Bundle` directly:
```go
type Bundle struct {
	// BaseCRL is the parsed base CRL
	BaseCRL *x509.RevocationList

	// DeltaCRL is the parsed delta CRL
	//
	// TODO: support delta CRL https://github.com/notaryproject/notation-core-go/issues/228
	// It will always be nil until we support delta CRL
	DeltaCRL *x509.RevocationList
}
```

We may need another design for OCSP cache.

Reply:
Added a issue https://github.com/notaryproject/notation-core-go/issues/230